core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/util/EditorUtility.java

```diff
@@ -676,8 +676,20 @@ public static String getEditorID(IEditorInput input, Object inputObject) {
 				contentType = Platform.getContentTypeManager().getContentType(CCorePlugin.CONTENT_TYPE_BINARYFILE);
 			}
 		}
-		IEditorRegistry registry = PlatformUI.getWorkbench().getEditorRegistry();
-		IEditorDescriptor desc = registry.getDefaultEditor(input.getName(), contentType);
+		IEditorDescriptor desc = null;
+		if (input instanceof ExternalEditorInput externalInput) {
```

Comment:
IMO the risk here is moderate, because this code will be reached only if the input is not  an instance of `IFileEditorInput` and it must be an instanceof `ExternalEditorInput` which limits the call on non workspace files as well. However, when this code should throw an exception, the previous behavior is ensured (desc is still null).

Reply:
Can you have a look at https://github.com/eclipse-cdt/cdt/blob/46d7fb74da53129d42296ef3f86730ddd811202e/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/CDebugModelPresentation.java#L232-L240 and let me know if that pattern (which is also used in LSP4E) is more general and applicable here? 

Besides that question - I agree this change is low risk and worth getting in.

Reply:
yes, it works as well. Good point!